### ISSUE TITLE
Nested where-clause fails with "does not exist in the current database" when relationJoins is enabled

### ISSUE THREAD

**Issue by DerJacques** [prisma/prisma#25103]
### Bug description

After upgrading to version 5.19.0, one of our queries has started to fail with the following error:

```
The column `t70.deletedAt` does not exist in the current database.
```

The column that's referenced in the error _does_ in fact exist in the database.
Disabling the `relationJoins` preview feature resolves the issue, and the query works as expected. The Typescript types also don't complain about the attribute.

I noticed that the release notes reference the following issue, which references the same error message:
https://github.com/prisma/prisma/issues/23742

### How to reproduce

Using the `contactDetailsIncludeInput` described below will cause the issue to occur on our end.
Removing the `deletedAt: null` condition resolves the error.

### Prisma information

```ts
export const contactDetailsIncludeInput = {
  identities: {
    include: {
      subscriptions: {
        where: {
          optedOutAt: null,
          audience: {
            deletedAt: null  <------- This is the one Prisma complains about
          }
        },
        include: {
          audience: true
        }
      }
    }
  }
} satisfies Prisma.ContactInclude
```

### Environment & setup

- OS: macOS
- Database: Postgres
- Node.js version: v20.16.0

### Prisma Version

```
5.19.0
```

**Comment by callumJPritchard** [comment]
I get a very similar issue with previously working code, when moving to 5.19.0

I've renamed some stuff here but the structure is identical to the part of the seed script that fails

```
return prisma4.modelName.findUniqueOrThrow(
The column `t15.fieldName` does not exist in the current database
```
the code causing this is:

```
prisma.modelName.findUniqueOrThrow({
        where: { id },
        include: {
            relatedModel: {
                where: {
                    type: {
                        fieldName: false
                    }
	...
```

**Comment by jkomyno** [comment]
I've actually just reproduced the issue.

# Setup

- `schema.prisma`
  ```prisma
  generator client {
    provider        = "prisma-client-js"
    previewFeatures = ["relationJoins"]
  }
  
  datasource db {
    provider  = "postgresql"
    url       = env("TEST_POSTGRES_URI")
  }
  
  model Contact {
    id         String      @id @default(cuid())
    identities Identity[]
  }
  
  model Identity {
    id              String         @id @default(cuid())
    contactId       String
    contact         Contact        @relation(fields: [contactId], references: [id])
    subscriptions   Subscription[]
  }
  
  model Subscription {
    id          String    @id @default(cuid())
    identityId  String
    audienceId  String
    optedOutAt  DateTime?
    audience    Audience  @relation(fields: [audienceId], references: [id])
    identity    Identity  @relation(fields: [identityId], references: [id])
  }
  
  model Audience {
    id         String     @id @default(cuid())
    deletedAt  DateTime?
    subscriptions Subscription[]
  }
  ```

- `seed.ts`
  ```typescript
  import { PrismaClient } from '.prisma/client'
  
  const prisma = new PrismaClient()
  
  async function main() {
    // Create some sample audiences
    const audience1 = await prisma.audience.create({
      data: {
        id: 'audience1',
        deletedAt: null,
      },
    })
  
    const audience2 = await prisma.audience.create({
      data: {
        id: 'audience2',
        deletedAt: null,
      },
    })
  
    // Create a contact with identities and subscriptions
    const contact = await prisma.contact.create({
      data: {
        id: 'contact1',
        identities: {
          create: [
            {
              id: 'identity1',
              subscriptions: {
                create: [
                  {
                    id: 'subscription1',
                    audienceId: audience1.id,
                    optedOutAt: null,
                  },
                  {
                    id: 'subscription2',
                    audienceId: audience2.id,
                    optedOutAt: null,
                  },
                ],
              },
            },
          ],
        },
      },
    })
  
    console.log({ contact })
  }
  
  main()
    .then(async () => {
      await prisma.$disconnect()
    })
    .catch(async (e) => {
      console.error(e)
      await prisma.$disconnect()
      process.exit(1)
    })
  ```
- index.ts
  ```typescript
  import { PrismaClient, Prisma } from '.prisma/client'
  
  const prisma = new PrismaClient()
  
  async function main() {
    const contactDetailsIncludeInput = {
      identities: {
        include: {
          subscriptions: {
            where: {
              optedOutAt: null,
              audience: {
                deletedAt: null,
              },
            },
            include: {
              audience: true,
            },
          },
        },
      },
    } satisfies Prisma.ContactInclude
  
    const contacts = await prisma.contact.findMany({
      include: contactDetailsIncludeInput,
    })
  
    console.log('contacts')
    console.dir(contacts, { depth: null })
  }
  
  main()
    .then(async () => {
      await prisma.$disconnect()
    })
    .catch(async (e) => {
      console.error(e)
      await prisma.$disconnect()
      process.exit(1)
    })
  ```

- package.json
  ```
  {
    "private": true,
    "name": "reprod-25103",
    "main": "index.ts",
    "scripts": {
      "start": "ts-node index.ts"
    },
    "dependencies": {
      "@prisma/client": "5.19.0"
    },
    "devDependencies": {
      "prisma": "5.19.0",
      "ts-node": "10.9.1",
      "typescript": "5.5.0"
    },
    "prisma": {
      "seed": "ts-node ./seed.ts"
    }
  }
  ```

# Commands

- Push schema to the database
  ```sh
  ❯ pnpm prisma db push
  ```

- Seed example data into the database
  ```sh
  ❯ pnpm prisma db seed
  ```

- Run the `index.ts` snippet
  ```sh
  ❯ pnpm start
  ```

  Output:

  ```sh
  PrismaClientKnownRequestError: 
  Invalid `prisma.contact.findMany()` invocation in
  /Users/jkomyno/work/prisma/prisma/sandbox/basic-postgres/index.ts:24:41
  
    21   },
    22 } satisfies Prisma.ContactInclude;
    23 
  → 24 const contacts = await prisma.contact.findMany(
  The column `t7.deletedAt` does not exist in the current database.
      at Ln.handleRequestError (/Users/jkomyno/work/prisma/prisma/sandbox/basic-postgres/node_modules/.prisma/client/src/runtime/RequestHandler.ts:236:17)
      at Ln.handleAndLogRequestError (/Users/jkomyno/work/prisma/prisma/sandbox/basic-postgres/node_modules/.prisma/client/src/runtime/RequestHandler.ts:176:12)
      at Ln.request (/Users/jkomyno/work/prisma/prisma/sandbox/basic-postgres/node_modules/.prisma/client/src/runtime/RequestHandler.ts:144:12)
      at async l (/Users/jkomyno/work/prisma/prisma/sandbox/basic-postgres/node_modules/.prisma/client/src/runtime/getPrismaClient.ts:932:69)
      at async main (/Users/jkomyno/work/prisma/prisma/sandbox/basic-postgres/index.ts:24:20) {
    code: 'P2022',
    clientVersion: '0.0.0',
    meta: { modelName: 'Contact', column: 't7.deletedAt' }
  }
  ```

# Expectation

If we remove `previewFeatures = ["relationJoins"]` from `schema.prisma` and re-run the same steps, we get:

```js
[
  {
    id: 'contact1',
    identities: [
      {
        id: 'identity1',
        contactId: 'contact1',
        subscriptions: [
          {
            id: 'subscription1',
            identityId: 'identity1',
            audienceId: 'audience1',
            optedOutAt: null,
            audience: { id: 'audience1', deletedAt: null }
          },
          {
            id: 'subscription2',
            identityId: 'identity1',
            audienceId: 'audience2',
            optedOutAt: null,
            audience: { id: 'audience2', deletedAt: null }
          }
        ]
      }
    ]
  }
]
```

**Comment by jkomyno** [comment]
Note: this regression was introduced in https://github.com/prisma/prisma-engines/pull/4968, which in turn was an attempt at fixing https://github.com/prisma/prisma/issues/23742.

**Comment by khalilsarwari** [comment]
also having this issue with nested query with relation join set

**Comment by apolanc** [comment]
This should now be fixed in the `5.19.1` version, closing this now.
